core/base/identifiers/Identifiers.h

```diff
@@ -0,0 +1,959 @@
+///
+/// \ingroup base
+/// \class ttk::Identifiers
+/// \author Eve Le Guillou <eve.le-guillou@lip6.fr>
+/// \date October 2022.
+///
+/// This module defines the %Identifiers class that computes for each vertex of
+/// a triangulation the average scalar value of itself and its direct neighbors.
+///
+///
+
+#pragma once
+
+// ttk common includes
+#include <Debug.h>
+#ifdef TTK_ENABLE_MPI
+#include <Geometry.h>
+#include <KDTree.h>
+#include <map>
+#endif
+
+namespace ttk {
+
+#ifdef TTK_ENABLE_MPI
+  struct Point {
+    double x;
+    double y;
+    double z;
+    ttk::SimplexId localId;
+  };
+
+  struct Response {
+    ttk::SimplexId id;
+    ttk::SimplexId globalId;
+  };
+
+#endif
+
+  /**
+   * The Identifiers class provides methods to compute for each vertex and
+   * cell of a data set a global id. It uses RankArray and outdated global
+   * identifiers if they are defined to accelerate computation for data sets
+   * of PolyData type.
+   */
+  class Identifiers : virtual public Debug {
+
+  public:
+    Identifiers();
+
+    ~Identifiers() override = default;
+
+  protected:
+    ttk::SimplexId vertexNumber_{};
+    ttk::SimplexId cellNumber_{};
+    std::vector<ttk::SimplexId> *vertexIdentifiers_;
+    std::vector<ttk::SimplexId> *cellIdentifiers_;
+#ifdef TTK_ENABLE_MPI
+    std::map<ttk::SimplexId, ttk::SimplexId> vertGtoL_;
```

Comment:
the (abstract) triangulation class already contains such a map (`vertexGidToLid_`).
couldn't you use this map instead? (that way it will be filled once for all -- currently, I have the impression that information is filled twice: once here, once letter when attaching the result of this computation to the triangulation).

Reply:
Here, the triangulation is not created yet, therefore vertexGidToLid_ doesn't exist.
Indeed, the map is created twice, but I don't know how to transfer the map from identifier to a triangulation that isn't created yet.

Reply:
see 4th point in the comments below.